_studio/shared/src/libmfx_core_hw.cpp

```diff
@@ -0,0 +1,114 @@
+#include "libmfx_core_hw.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <string>
+#include <list>
+#include <fstream>
+#include <iomanip>
+
+const std::string PCI_DIR = "/sys/bus/pci/devices";
+constexpr mfxU32 PCI_DISPLAY_CONTROLLER_CLASS = 0x03;
+constexpr mfxU32 CLASS_ID_OFFSET = 16;
+constexpr mfxU32 INTEL_VENDOR_ID = 0x8086;
+
+int mfx_dir_filter(const struct dirent* dir_ent)
+{
+    if (!dir_ent) return 0;
+    if (!strcmp(dir_ent->d_name, ".")) return 0;
+    if (!strcmp(dir_ent->d_name, "..")) return 0;
+    return 1;
+}
+
+mfxU32 ReadFirstHexNumInFile(std::string file_name)
```

Comment:
Also would be used only on Linux... Either should be moved under define (and declared static) or done via lambda in a function where you really use it.